### PR TITLE
[air] Dreambooth finetuning workspace template

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

This makes the dreambooth fine-tuning example a workspace template. This mostly adds a README, and moves the `pip install` out of the script (since it will already be packaged in a docker image).

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/templates/05_dreambooth_finetuning/configs/compute/gce.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# 4 A100 GPUs with 40gb GRAM each
+# This is a bit overkill, but instances with L4/A10G GPUs are not yet available on GCE
+head_node_type:
+  name: head_node_type
+  instance_type: a2-highgpu-2g-nvidia-a100-40gb-4
```
Comment (justinvyu) [inline doc/source/templates/05_dreambooth_finetuning/configs/compute/gce.yaml:5]: Do we know when these other instance types be available?
There's some more work to be done to get this example running on a single GPU (using deepspeed / gradient checkpointing / some other tricks). Let's do that in a follow-up.
 
https://cloud.google.com/compute/docs/gpus#l4-gpus

File: doc/source/ray-air/examples/dreambooth_finetuning.rst [hunk 2]
```diff
@@ -23,7 +23,6 @@ This example fine-tunes both the ``text_encoder`` and ``unet`` models used in th
 
 
 .. image:: images/dreambooth_example.png
```
Comment (matthewdeng) [inline doc/source/ray-air/examples/dreambooth_finetuning.rst:25]: Some files moved:

https://buildkite.com/ray-project/oss-ci-build-pr/builds/29778#01899675-7970-4332-b685-2629d6892c58
```
/ray/doc/source/ray-air/examples/dreambooth_finetuning.rst:: WARNING: image file not readable: ray-air/examples/images/dreambooth_example.png
/ray/doc/source/ray-air/examples/dreambooth_finetuning.rst:: WARNING: image file not readable: ray-air/examples/images/dreambooth_training.png
```

File: doc/source/ray-air/examples/dreambooth_finetuning.rst [hunk 3]
```diff
@@ -23,7 +23,6 @@ This example fine-tunes both the ``text_encoder`` and ``unet`` models used in th
...
-   :target: images/dreambooth_example.png
    :alt: DreamBooth example
 
 The full code repository can be found here: `https://github.com/ray-project/ray/blob/master/python/ray/air/examples/dreambooth/ <https://github.com/ray-project/ray/blob/master/python/ray/air/examples/dreambooth/>`_
```
Comment (matthewdeng) [inline doc/source/ray-air/examples/dreambooth_finetuning.rst:28]: I think this will need to point to the new template directory?

**Review by matthewdeng — APPROVED** [review]

File: doc/source/ray-air/examples/dreambooth_finetuning.rst (hunk 3 above)
Comment (justinvyu) [reply doc/source/ray-air/examples/dreambooth_finetuning.rst:28]: The images are under this `doc/source/ray-air/examples` folder though, not under the template directory. Very weird 😨

Comment (justinvyu) [reply doc/source/ray-air/examples/dreambooth_finetuning.rst:28]: I'll try an absolute path from source dir

Comment (justinvyu) [reply doc/source/ray-air/examples/dreambooth_finetuning.rst:28]: Oh oops nevermind, the images were in the template directory.
